doc/thingpedia-nl-support.md

```diff
@@ -67,10 +154,80 @@ action share(in req status: String #_[prompt="What do you want to post?"])
+Examples can be found in all the tutorials. 
+
+Note that the confirmation sentence should include all the __required__ input parameter
+with `$param` or `${param}`.
 
 ### Output format
 When a stream or a query is combined with the action `notify`, 
 the returned results will be presented to the user in Almond. 
 By default, Almond will simply list the value of all the output parameters one by one 
-in the order they are declared in the function signature.
\ No newline at end of file
+in the order they are declared in the function signature.
+
+You can customize the output format with the `formatted` annotation.
+The annotation takes a list of Objects. 
+Each Object represents one response and each response will be presented to users in the same order in the list. 
+Four types of responses are supported: `text`, `picture`, `rdl`, and `code`.
+
+#### Text response 
+A text response returns a simple text message.
+It has only one property: `text`. 
+E.g., `{ type="text",text="Works in ${industry}" }` (in [Tutorial 3](/doc/thingpedia-tutorial-linkedin.md)).
+
+####  Picture response: 
+A picture response shows a picture to the user. 
+It has only one property: `url`.
+E.g., `{ type: "picture", url: "${picture_url}" }` (in [Tutorial 3](/doc/thingpedia-tutorial-linkedin.md)).
+
+#### RDL response: 
+An RDL response returns a clickable with a title and a description, suitable for website 
+links and news articles.
+It has three properties: `webCallback` (the link), `displayTitle` (the title), 
+and `displayText` (the description, optional). 
+See [Tutorial 1](/doc/thingpedia-tutorial-nyt.md) for an example of this format type.
+
+#### Customized response
+If you need more control over the output, such as different output based 
+on results, you can choose `code` type and write Javascript code in the `code` property. 
+The code contains a [self-executing anonymous function](https://developer.mozilla.org/en-US/docs/Glossary/IIFE)
+in String format.
+The function will be invoked with three arguments: 
+the result of your function (an object with each parameter as a property), 
+a hint informing you of how the result will be shown to the user, 
+and a `Formatter` object. 
+The function can return a string, a formatted message object 
+(with the same structure as the JSON described here) 
+or an array of objects.
+
+The `Formatter` object provides the following methods for your convenience:
+- `measureToString(value, precision, unit)`: convert a value of `Measure` type to a string.
+- `dataToString(date)`: convert a value of `Date` type to a string containing only the date.
+- `timeToString(date)`: convert a value of `Date` type to a string containing only the time.
+- `dateAndTimeToString(date)`: convert a value of `Date` type to a string containing both the date and the time.
+- `locationToString(date)`: convert a value of `Location` type to a string
+- `anyToString(object)`: convert any other value to a string (unnecessary for `String` type)
```

Comment:
You should use the methods of `Formatter` rather than the equivalent native JS methods (such as `date.toLocaleString()`) to respect the user's setting of locale and timezone.